src/VisualStudio/Core/Def/Implementation/SolutionSize/SolutionSizeTracker.cs

```diff
@@ -0,0 +1,186 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Concurrent;
+using System.Composition;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.SolutionCrawler;
+using Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.SolutionSize
+{
+    /// <summary>
+    /// Track approximate solution size.
+    /// </summary>
+    [Export]
+    [ExportIncrementalAnalyzerProvider(WorkspaceKind.Host), Shared]
+    internal class SolutionSizeTracker : IIncrementalAnalyzerProvider
+    {
+        private readonly IncrementalAnalyzer tracker = new IncrementalAnalyzer();
+
+        public long GetSolutionSize(Workspace workspace, SolutionId solutionId)
+        {
+            var vsWorkspace = workspace as VisualStudioWorkspaceImpl;
+            if (vsWorkspace == null)
+            {
+                return -1;
+            }
+
+            return tracker.GetSolutionSize(solutionId);
```

Comment:
Even if you do not want to convert `if` into `? :`, the usage of `as` seems unnecessary. Just use `is`.